packages/webpack-cli/lib/webpack-cli.js

```diff
@@ -373,8 +373,11 @@ class WebpackCLI {
             {
                 name: 'env',
                 type: (value, previous = {}) => {
+                    // for https://github.com/webpack/webpack-cli/issues/2642
+                    const regExpForSplitting = (value.match(/=/g) || []).length === 1 ? /=/ : /=(.+)/;
```

Comment:
why?

Reply:
When passing `--env foo=""` via terminal the value is actually `foo=` and not `foo=""` hence setting `env.foo=` = `true`,

I don't think it is problem with our CLI, the terminal just ignores `""`.

To get the desired behavior we need to pass `--env foo=\"\"` with this the value is `foo=""` and hence the desired behavior.

Reply:
I am not sure if we should accept this change it is more of a workaround for the situation. Maybe we can mention this somewhere in docs and leave it as is.

Reply:
There is logical error, for any value containing = this will match, better to look for last char in the string and if it's `=` then set value as `""`

Reply:
Can you give example?

Reply:
yep, look how it uses first regex in all cases

![image](https://user-images.githubusercontent.com/21009455/115150581-9f15cd00-a086-11eb-9db9-647dfe82e0a4.png)

Reply:
For any value containing only one `=` it will match and the corresponding regexp will parse it correctly. I believe test cases are already in place. Am I missing something here?

Reply:
maybe we are good with this, I'm not sure why the other was used, but there was some suggestion on the PR, try with this regex in all cases, we might be good with this (`/=/`)

Reply:
i.e remove the regexExpForSplitting logic

Reply:
I will improve and add more tests 👍

Reply:
💯 💯 💯

Reply:
Other is for the case of multiple `=`. 

`--env foo=bar=app` => `foo: 'bar=app'`

Reply:
hmm yes, what about cases when `foo=bar=""` quite absurd but possibility

Reply:
Updated 👍🏻